File: 05. ObjectOrientedProgramming/04. EncapsulationAndPolymorphism/01. Shapes/Rectangle.cs
using System;

class Rectangle : BasicShape
{
    public Rectangle(double width, double height)
        : base(width, height)
    {
    }

    public override double CalculateArea()
    {
        return base.Height * base.Width;
    }

    public override double CalcualtePerimiter()
    {
        return 2 * base.Width + 2 * base.Height;
    }

    public override string ToString()
    {
        return string.Format("Width: {0:N2}, Height: {1:N2}, Area: {2:N2}, Perimiter: {3:N2}",
            base.Width, base.Height, this.CalculateArea(), this.CalcualtePerimiter());
    }

}

File: 05. ObjectOrientedProgramming/09. OOPinJava/src/_01/Geometry/Rectangle.java
package _01.Geometry;

public class Rectangle extends PlaneShape {
	private double width;
	private double height;
	
	public Rectangle(Vertex[] vertices, double width, double height) {
		super(vertices);
		this.setWidth(width);
		this.setHeight(height);
	}
	
	public double getWidth() {
		return width;
	}
	
	public void setWidth(double width) {
		this.width = width;
	}
	
	public double getHeight() {
		return height;
	}
	
	public void setHeight(double height) {
		this.height = height;
	}
	
	@Override
	public double getArea() {
		double area = this.width * this.height;
		
		return area;
	}
	
	@Override
	public double getPerimeter() {
		double perimeter = this.width * 2 + this.height * 2;
		
		return perimeter;
	}
	
	@Override
	public String toString(){
		return "Type: " + this.getClass() + 
			   ". Center:" + super.getVertices()[0] +
			   " has perimeter: " + this.getPerimeter() +
			   " and area: " + this.getArea();
	}
	
}
